cpp/arcticdb/storage/s3/s3_storage.cpp

```diff
@@ -32,6 +34,48 @@ std::streamsize S3StreamBuffer::xsputn(const char_type* s, std::streamsize n) {
 }
 }
 
+void check_ec2_metadata_endpoint(const S3ApiInstance& s3_api) {
+    auto disabled_env = Aws::Environment::GetEnv("AWS_EC2_METADATA_DISABLED");
+    if (Aws::Utils::StringUtils::ToLower(disabled_env.c_str()) == "true") {
+        const char* who_disabled = s3_api.is_ec2_metadata_disabled_by_us() ?
+            "EC2 metadata endpoint did not respond in time so was bypassed":
+            "AWS_EC2_METADATA_DISABLED environment variable is set to true";
```

Comment:
Is this the wrong way round?

Reply:
It is the correct way around.

When `is_ec2_metadata_disabled_by_us()` is true, how we disabled it (by setting `AWS_EC2_METADATA_DISABLED` internally) is an implementation detail that we can change at any time, so we only need to let the user know the relevant downside.

When the user disables it, s/he should be familiar with the environment name already, so we can be explicit.